### PR TITLE
feat(rules): Add no-empty-title rule

### LINKED ISSUE #226: New rule request: no-empty-title
Having an empty string as title is useless. Can we have a rule that prevents this?

### PR DESCRIPTION
**Pull request by himynameisdave**
Having an empty string as title is pretty useless. This adds a new rule to ensure that tests have real titles.

Resolves #226

---

The following patterns are considered warnings:

```js
describe('', () => {});
describe('foo', () => {
  it('', () => {});
});
it('', () => {});
test('', () => {});
xdescribe('', () => {});
xit('', () => {});
xtest('', () => {});
```

These patterns would not be considered warnings:

```js
describe('foo', () => {});
describe('foo', () => {
  it('bar', () => {});
});
test('foo', () => {});
it('foo', () => {});
xdescribe('foo', () => {});
xit('foo', () => {});
xtest('foo', () => {});
```

### CODE REVIEW

**Comment by himynameisdave** [comment]
@SimenB yes I can add to the `README`, and yes I can get us to 100% coverage!

**Comment by himynameisdave** [comment]
@SimenB resolved your comments.

Note [the build is still failing](https://travis-ci.org/jest-community/eslint-plugin-jest/jobs/504960882), but it is [also failing on `master`](https://travis-ci.org/jest-community/eslint-plugin-jest/jobs/504211344). 

Let me know when that gets resolved upstream and I'll merge that fix in here. Otherwise just let me know if you have any other comments to address! 👍

File: README.md [hunk 1]
```diff
@@ -96,6 +96,7 @@ for more information about extending configuration files.
 | [lowercase-name][]           | Disallow capitalized test names                                   |                  | ![fixable-green][]  |
 | [no-alias-methods][]         | Disallow alias methods                                            | ![recommended][] | ![fixable-green][]  |
 | [no-disabled-tests][]        | Disallow disabled tests                                           | ![recommended][] |                     |
+| [no-empty-title][]           | Disallow disabled tests                                           | ![recommended][] |                     |
```
Comment (SimenB) [inline README.md]: copy paste error in the description 🙂

**Comment by SimenB** [comment]
master is fixed, so a rebase/merge should do the trick 🙂

**Comment by himynameisdave** [comment]
@SimenB resolved the typo, let me know if I need to fix anything else!

**Comment by SimenB** [comment]
Added a test for template literals - if CI is happy then I'm happy 🙂

**Comment by himynameisdave** [comment]
@SimenB yeah it wasn't handling template literals well.

These cases weren't reporting:

```jsx
it(``, () => {});
it(`${someVar}`, () => {});
```

They are now though, so everything should work! Let me know if you need to me change anything else!

File: rules/util.js [hunk 2]
```diff
@@ -218,6 +218,7 @@ module.exports = {
   getStringValue,
   isDescribe,
   isFunction,
+  isTemplateLiteral,
```
Comment (himynameisdave) [inline rules/util.js]: whoops my bad!

**Comment by garyking** [comment]
The rule doesn't seem to work. It needs to be added to `index.js` right?

**Comment by SimenB** [comment]
Good point!

**Comment by himynameisdave** [comment]
Oops my bad!

**Comment by SimenB** [comment]
Fixed in 22.4.1, thanks for reporting @garyking!

**Comment by iamnewton** [comment]
Was this rule removed at some point?  I don't see it in the docs, nor in the file system and I'm not seeing reference to its removal.

**Comment by SimenB** [comment]
Merged into [`valid-title`](https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/valid-title.md) in bb2d09ea2084c12d6dc0b6fdb88c1fa8fa897273

**Comment by iamnewton** [comment]
Thanks @SimenB

**Pull request merged**